Updated the IIS rewrites article for V9 (#3482)
umbraco/UmbracoDocs: Reference/Routing/IISRewriteRules/index-v9.md

> ---
> versionFrom: 9.0.0
> ---
> 
> # Rewrites in Umbraco 9
> 
> With the release of Umbraco 9 and the change of the underlying framework to .NET Core 5 they way that you use rewrites have changed as well.
> 
> The IIS module have been replaced with [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0) for rewritting in Umbraco 9.
> This it means that there is no Web.config to add your rewrites in any more.
> 
> ## Enabling the rules
> 
> To use Rewrites with Umbraco 9 you need to create an XML file with your rules and register it in your `Startup.cs` in your project by creating an instance of the `RewriteOptions` class with extension methods for each of your rewrite rules.

Author: ```suggestion
---
versionFrom: 9.0.0
---

# Rewrites in Umbraco 9

With the release of Umbraco 9 and the change of the underlying framework to .NET Core 5, the way that you use rewrites has changed as well.

The IIS module have been replaced with [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0) for rewriting in Umbraco 9.
This means that there is no Web.config to add your rewrites in anymore.

## Enabling the rules

To use Rewrites with Umbraco 9 you need to create an XML file with your rules and register it in your `Startup.cs` in your project by creating an instance of the `RewriteOptions` class with extension methods for each of your rewrite rules.
```

> ### Example
> 
> An example of how this can be done is
> 
> - Create an XML file with your rewrites in and place it in your project:
> 
> ```xml
> <?xml version="1.0" encoding="utf-8" ?>
> <rewrite>
>   <rules>
>     <rule name="Redirects umbraco.io to actual domain" stopProcessing="true">
>       <match url=".*" />
>       <conditions>
>         <add input="{HTTP_HOST}" pattern="^(.*)?.euwest01.umbraco.io$" />
>         <add input="{REQUEST_URI}" negate="true" pattern="^/umbraco" />
>         <add input="{REQUEST_URI}" negate="true" pattern="^/DependencyHandler.axd" />
>         <add input="{REQUEST_URI}" negate="true" pattern="^/App_Plugins" />
>         <add input="{REQUEST_URI}" negate="true" pattern="localhost" />
>       </conditions>
>       <action type="Redirect" url="http://jonathanpabst.com>.com/{R:0}"
>         appendQueryString="true" redirectType="Permanent" />
>     </rule>
>   </rules>
> </rewrite>
> ```
> 
> - With the XML file with the rules created, you need to create an instance of the RewriteOptions class with your extension methods for each rule in the Startup.cs file:
> 
> ```Csharp
>         public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
>         {
>             using (StreamReader iisUrlRewriteStreamReader = File.OpenText("Umbraco/Rewrites.xml"))
>             {
>                 var options = new RewriteOptions()
>                     .AddIISUrlRewrite(iisUrlRewriteStreamReader);
>                 app.UseRewriter(options);
>             }
> ```
> 
> For more in-depth information on how to use rewrites in Umbraco 9, check out the documentation for [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0)

Author: ```suggestion
### Example

An example of how this can be done is

- Create an XML file with your rewrites in and place it in your project:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<rewrite>
  <rules>
    <rule name="Redirects umbraco.io to actual domain" stopProcessing="true">
      <match url=".*" />
      <conditions>
        <add input="{HTTP_HOST}" pattern="^(.*)?.euwest01.umbraco.io$" />
        <add input="{REQUEST_URI}" negate="true" pattern="^/umbraco" />
        <add input="{REQUEST_URI}" negate="true" pattern="^/DependencyHandler.axd" />
        <add input="{REQUEST_URI}" negate="true" pattern="^/App_Plugins" />
        <add input="{REQUEST_URI}" negate="true" pattern="localhost" />
      </conditions>
      <action type="Redirect" url="http://jonathanpabst.com>.com/{R:0}"
        appendQueryString="true" redirectType="Permanent" />
    </rule>
  </rules>
</rewrite>
```

- With the XML file with the rules created, you need to create an instance of the RewriteOptions class with your extension methods for each rule in the Startup.cs file:

```Csharp
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            using (StreamReader iisUrlRewriteStreamReader = File.OpenText("Umbraco/Rewrites.xml"))
            {
                var options = new RewriteOptions()
                    .AddIISUrlRewrite(iisUrlRewriteStreamReader);
                app.UseRewriter(options);
            }
```

For a more in-depth information on how to use rewrites in Umbraco 9, check out the documentation for [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0)
```

---
versionFrom: 9.0.0
---

# Rewrites in Umbraco 9

> With the release of Umbraco 9 and the change of the underlying framework to .NET Core 5, the way that you use rewrites has changed as well.

Author: ```suggestion
With the release of Umbraco 9 and the change of the underlying framework to .NET 5, the way that you use rewrites has changed as well.
```

> The IIS module have been replaced with [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0) for rewriting in Umbraco 9.

Author: ```suggestion
The URL Rewriting extension in IIS has been replaced with [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-5.0) for rewriting in Umbraco 9.
```
